psutil/arch/windows/process_info.c

```diff
@@ -804,10 +870,78 @@ psutil_get_cmdline(long pid) {
     PyObject *py_unicode = NULL;
     LPWSTR *szArglist = NULL;
     int nArgs, i;
-
-    if (psutil_get_process_data(pid, KIND_CMDLINE, &data, &size) != 0)
-        goto out;
-
+    HANDLE hProcess;
+    int windows_version;
+    ULONG ret_length = 4096;
+    NTSTATUS status;
+    char * cmdline_buffer = NULL;
+    WCHAR * cmdline_buffer_wchar = NULL;
+    PUNICODE_STRING tmp = NULL;
+    DWORD string_size;
+    _NtQueryInformationProcess NtQueryInformationProcess = NULL;
+
+    windows_version = get_windows_version();
+    if (windows_version >= WINDOWS_81) {
+        // by defaut, still use PEB (if command line params have been patched in PEB, we will get the actual ones)
+        if (psutil_get_process_data(pid, KIND_CMDLINE, &data, &size) != 0) {
+            PyErr_Clear(); // reset that we had an error, and retry with NtQueryInformationProcess
+            // if it fails, fallback to NtQueryInformationProcess (for protected processes)
+            NtQueryInformationProcess = get_NtQueryInformationProcess();
+            if (NtQueryInformationProcess != NULL) {
```

Comment:
In order to stay consistent with the rest of the code I suggest to check for failure instead of success. This way we will avoid long `if` blocks and indentation. Also avoid `else` at all (you save one indented block). In practice: 

```c
if (NtQueryInformationProcess == NULL) {
    PyErr_SetFromWindowsErr(0);
    goto out;
}
...
```